src/mlpack/tests/linear_regression_test.cpp

```diff
@@ -11,20 +11,18 @@
 #include <mlpack/core.hpp>
 #include <mlpack/methods/linear_regression/linear_regression.hpp>
 
-#include <boost/test/unit_test.hpp>
-#include "test_tools.hpp"
-#include "serialization.hpp"
+#include "serialization_catch.hpp"
+#include "test_catch_tools.hpp"
+#include "catch.hpp"
 
 using namespace mlpack;
 using namespace mlpack::regression;
 
-BOOST_AUTO_TEST_SUITE(LinearRegressionTest);
-
 /**
  * Creates two 10x3 random matrices and one 10x1 "results" matrix.
  * Finds B in y=BX with one matrix, then predicts against the other.
  */
-BOOST_AUTO_TEST_CASE(LinearRegressionTestCase)
+TEST_CASE("LinearRegressionTestCase", "[LinearRegressionTest]")
```

Comment:
```suggestion
TEST_CASE("LinearRegressionTestCase", "LinearRegressionTest")
```
I'm not sure if the `[` is useful---maybe I overlooked something?  At least in the activation functions test, I didn't use it.

Reply:
It's probably the same, but since in the documentation https://github.com/catchorg/Catch2/blob/master/docs/test-cases-and-sections.md#tags the `[]` it might be a good idea to go with the style. Also, we use the same style in ensmallen as well.

Also, what about we introduce two more tags to distinguish between bindings and other tests?

Reply:
Sure, that works for me. :+1: